### PR TITLE
Add interface names into NSM metrics

### PR DESCRIPTION
**Pull request by NikitaSkrynnik**
## Issue
https://github.com/networkservicemesh/sdk-vpp/issues/768

## Original PR
https://github.com/networkservicemesh/sdk/pull/1560

## How Has This Been Tested?

- [x] Added unit testing to cover
- [ ] Tested manually
- [ ] Tested by integration testing
- [ ] Have not tested

## Types of changes

- [ ] Bug fix
- [x] New functionality
- [ ] Documentation
- [ ] Refactoring
- [ ] CI

### CODE REVIEW

File: pkg/networkservice/common/mechanisms/client.go [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// Copyright (c) 2021 Doc.ai and/or its affiliates.
+// Copyright (c) 2021-2023 Doc.ai and/or its affiliates.
```
Comment (VitalyGushin) [inline pkg/networkservice/common/mechanisms/client.go:1]: 2024

**Comment by VitalyGushin** [comment]
LGTM

**Review by VitalyGushin — APPROVED** [review]

File: pkg/networkservice/chains/endpoint/combine_test.go [hunk 2]
```diff
@@ -83,14 +83,23 @@ func testCombine(t *testing.T, mechanism *networkservice.Mechanism) {
...
-			kernel.MECHANISM: servers[0],
-			memif.MECHANISM:  servers[1],
-		})
-	}, newTestEndpoint(ctx, kernel.MECHANISM), newTestEndpoint(ctx, memif.MECHANISM))
-
-	cc := startEndpoint(ctx, t, e)
+		return next.NewNetworkServiceServer(
+			begin.NewServer(),
+			metadata.NewServer(),
+			monitor.NewServer(ctx, &m),
```
Comment (denis-tingaikin) [inline pkg/networkservice/chains/endpoint/combine_test.go:91]: In most cases, we're using these elements inside endpoints that we're combining. So I'd prefer to rework the test to keep it similar to cmd-forwarder-vpp. See at https://github.com/networkservicemesh/cmd-forwarder-vpp/blob/main/internal/xconnectns/server.go#L92

File: pkg/networkservice/chains/endpoint/combine_test.go [hunk 3]
```diff
@@ -310,13 +313,15 @@ type testEndpoint struct {
...
-func newTestEndpoint(ctx context.Context, name string) *testEndpoint {
+func newTestEndpoint(ctx context.Context, mechanism, name string) *testEndpoint {
 	e := new(testEndpoint)
 	e.NetworkServiceServer = next.NewNetworkServiceServer(
-		updatepath.NewServer(name),
 		begin.NewServer(),
 		metadata.NewServer(),
 		monitor.NewServer(ctx, &e.MonitorConnectionServer),
+		mechanisms.NewServer(map[string]networkservice.NetworkServiceServer{
+			mechanism: updatepath.NewServer(name),
```
Comment (glazychev-art) [inline pkg/networkservice/chains/endpoint/combine_test.go:323]: Have you tried using null.NewServer() here and leaving updatePath at the previous place?

Comment (NikitaSkrynnik) [reply pkg/networkservice/chains/endpoint/combine_test.go:323]: Yes, I have. It didn't work

**Review by denis-tingaikin — APPROVED** [review]

**Pull request merged**